clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/ttl/VeniceRmdTTLFilter.java

```diff
@@ -105,15 +109,22 @@ public void close() {
 
   boolean filterByTTLandMaybeUpdateValue(final INPUT_VALUE value) {
     ByteBuffer rmdPayload = getRmdPayload(value);
-    if (rmdPayload == null || !rmdPayload.hasRemaining()) {
+    boolean hasRmdPayload = rmdPayload != null && rmdPayload.hasRemaining();
+    if (!hasRmdPayload && ttlPolicy == TTLResolutionPolicy.RT_WRITE_ONLY) {
       throw new IllegalStateException(
           "The record doesn't contain required RMD field. Please check if your store has A/A enabled");
     }
     int valueSchemaId = getSchemaId(value);
     int id = getRmdProtocolId(value);
     RmdVersionId rmdVersionId = new RmdVersionId(valueSchemaId, id);
-    GenericRecord rmdRecord =
-        rmdDeserializerCache.computeIfAbsent(rmdVersionId, this::generateRmdDeserializer).deserialize(rmdPayload);
+    final GenericRecord rmdRecord;
+    if (hasRmdPayload) {
+      rmdRecord =
+          rmdDeserializerCache.computeIfAbsent(rmdVersionId, this::generateRmdDeserializer).deserialize(rmdPayload);
+    } else {
+      long logicalTimestamp = getLogicalTimestamp(value);
```

Comment:
I guess the offset are used for comparison for values of same key in batch portion of data?
In case the offset grows higher than the time, what if we use one timestamp like start time of the push, and then decrement all the offsets based timestamp by that one timestamp value:
logicalTimestamp1 = offset1 - pushStartTime;
logicalTimestamp2 = offset2 - pushStartTime;

Reply:
Was due to confusion of offset and push timestamp. Updated